packages/beeq/src/components/page-title/bq-page-title.tsx

```diff
@@ -0,0 +1,126 @@
+import { Component, h, Prop } from '@stencil/core';
+
+/**
+ * @part wrapper - The wrapper container `<div>` of the element inside the shadow DOM
+ * @part back - The container `<div>` that wraps the page title back icon
+ * @part title - The container `<div>` that wraps the pate title content
+ * @part icon - The `<bq-icon>` element used to render a predefined icon size based on the page title (back, edit, download icon)
+ * @part actions - The container `<div>` element used to render a edit and download icons for page title
+ * @part sub-title - The container `<div>` that wraps the sub-title element
+ */
+@Component({
+  tag: 'bq-page-title',
+  styleUrl: './scss/bq-page-title.scss',
+  shadow: true,
+})
+export class BqPageTitle {
+  // Own Properties
+  // ====================
+
+  // Reference to host HTML element
+  // ===================================
+
+  // State() variables
+  // Inlined decorator, alphabetical order
+  // =======================================
+
+  // Public Property API
+  // ========================
+
+  // Prop lifecycle events
+  // =======================
+
+  /** If true, the page title back icon will be shown */
+  @Prop({ reflect: true }) showBackIcon: boolean;
+
+  /** If true, the page title back icon will be shown */
+  @Prop({ reflect: true }) showActionIcons: boolean;
+
+  /** If true, the sub title of page title will be shown */
+  @Prop({ reflect: true }) showSubTitle: boolean;
+
+  // Events section
+  // Requires JSDocs for public API documentation
+  // ==============================================
+
+  // Component lifecycle events
+  // Ordered by their natural call order
+  // =====================================
+
+  // Listeners
+  // ==============
+
+  // Public methods API
+  // These methods are exposed on the host element.
+  // Always use two lines.
+  // Public Methods must be async.
+  // Requires JSDocs for public API documentation.
+  // ===============================================
+
+  // Local methods
+  // Internal business logic.
+  // These methods cannot be called from the host element.
+  // =======================================================
+
+  // render() function
+  // Always the last one in the class.
+  // ===================================
+
+  render() {
+    return (
+      <div class="flex gap-xs px-xxl4 py-xl" part="wrapper">
+        {/* BACK ICON */}
+        <div class={{ 'flex p-s': true, hidden: !this.showBackIcon }} part="back">
+          <slot name="back">
+            <bq-icon
+              color="text--primary"
+              name="arrow-left"
+              size="24"
+              weight="bold"
+              part="icon"
+              exportparts="base,svg"
+            />
```

Comment:
The back navigation should handle accessibility, meaning this should be a button icon instead of just an icon.

Reply:
Also, in case consumers use the default back button navigation, we should provide a callback handler to allow consumers to trigger any type of action they would like to execute when this button is clicked, right?

Reply:
I think we should handle this as `prefix` / `suffix` like button.  We have [default title](https://www.figma.com/design/2VS47PTSkQWJkr7H6DpAdR/BEEQ-Design-System-(v1.0)?node-id=3991-104418&t=jiu69tRNYcC905Zl-0) without any adornments.

Reply:
Yes, you are right, but besides handling the prefix/suffix as you suggested, I'm also thinking of providing an easy way to use back navigation and save some time for developers to set up the layout. What do you guys think, @endv-bogdanb, @Cata1989